internal/resources/iampolicy/resource_iam_policy_test.go

```diff
@@ -0,0 +1,598 @@
+		ManagementClusterName:      managementClusterName,
+		ProvisionerName:            provisionerName,
+		NamespaceName:              namespaceName,
+		NamespaceRole1:             namespaceRole,
+		WorkspaceName:              workspaceName,
+		WorkspaceRole1:             workspaceRole,
+		KubeConfigPath:             os.Getenv("KUBECONFIG"),
+		ClusterName:                clusterName,
+		ClusterRole1:               clusterRole,
+		ClusterGroupNameForCluster: clusterGroupNameForCluster,
+		ClusterGroupName:           acctest.RandomWithPrefix(clusterGroupNamePrefix),
+		ClusterGroupRole1:          clusterGroupRole,
+		OrgID:                      os.Getenv("ORG_ID"),
+		OrgRole1:                   orgRole,
+		IAMPolicyResource:          IAMPolicyResource,
+		IAMPolicyResourceVar1:      IAMPolicyResourceVar1,
+		IAMPolicyResourceVar2:      IAMPolicyResourceVar2,
+		IAMPolicyResourceName1:     fmt.Sprintf("%s.%s", IAMPolicyResource, IAMPolicyResourceVar1),
+		IAMPolicyResourceName2:     fmt.Sprintf("%s.%s", IAMPolicyResource, IAMPolicyResourceVar2),
+		Subject1Name:               subject1Name,
+		Subject1Kind:               subject1Kind,
+	}
+}
+
+func TestAcceptanceForIAMPolicyResource(t *testing.T) {
+	testConfig := testGetDefaultAcceptanceConfig(t)
+
+	t.Log("start IAM policy resource acceptance tests!")
+
+	// Test case for single IAM policy resource.
+	resource.Test(t, resource.TestCase{
+		PreCheck:          testhelper.TestPreCheck(t),
+		ProviderFactories: testhelper.GetTestProviderFactories(testConfig.Provider),
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					if testConfig.OrgID == "" {
+						t.Skip("ORG_ID env var is not set for organization scoped IAM policy acceptance test")
+					}
+				},
+				Config: testConfig.getTestIAMPolicyResourceBasicConfigValue(organizationScope, false),
+				Check:  testConfig.checkIAMPolicyResourceAttributes(organizationScope),
+			},
+			{
+				Config: testConfig.getTestIAMPolicyResourceBasicConfigValue(clusterGroupScope, false),
+				Check:  testConfig.checkIAMPolicyResourceAttributes(clusterGroupScope),
+			},
+			{
+				PreConfig: func() {
+					if testConfig.KubeConfigPath == "" {
+						t.Skip("KUBECONFIG env var is not set for cluster scoped IAM policy acceptance test")
+					}
+				},
+				Config: testConfig.getTestIAMPolicyResourceBasicConfigValue(clusterScope, false),
+				Check:  testConfig.checkIAMPolicyResourceAttributes(clusterScope),
+			},
+			{
+				Config: testConfig.getTestIAMPolicyResourceBasicConfigValue(workspaceScope, false),
+				Check:  testConfig.checkIAMPolicyResourceAttributes(workspaceScope),
+			},
+			{
+				Config: testConfig.getTestIAMPolicyResourceBasicConfigValue(namespaceScope, false),
+				Check:  testConfig.checkIAMPolicyResourceAttributes(namespaceScope),
+			},
+		},
+	},
+	)
+
+	t.Log("IAM policy resource acceptance test complete for single resource!")
+	time.Sleep(2 * time.Minute)
```

Comment:
why sleep??

Reply:
For safety reasons, in case the resource is not deleted from the DB.

Reply:
two mins is a long time for a test to wait.

Reply:
If you want to wait for the resources to be cleaned then validate them through APIs, sleep is not the suggested way

Reply:
In the interest of time, can you added a TODO which let goes the hardcoded sleep and enhance the acceptance in a different MR. 
please merge the MR once the acceptance results are posted.